staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/builder/builder.go

```diff
@@ -74,6 +87,12 @@ type Options struct {
 	// Convert to OpenAPI v2.
 	V2 bool
 
+	// Only takes effect if the flag and V2 and both set to true. If the condition is reached,
+	// publish OpenAPI V2 but skip running the spec through ToStructuralOpenAPIV2
+	// This prevents XPreserveUnknownFields:true fields from being cleared
+	// Used only by server side apply
+	OverrideToStructuralOpenAPIV2 bool
```

Comment:
I'm not thrilled by a variable that strongly depends on another

Reply:
it's unfortunate how a lot of these options only make sense for v2 and or to fit things into v2 and server-side apply. Hopefully we can get rid of this when we deprecate v2.